src/runtime/utils/colors.ts

```diff
@@ -270,7 +278,10 @@ export const setGlobalColors = (theme: TWConfig['theme']) => {
     950: 'rgb(var(--color-gray-950) / <alpha-value>)'
   }
 
-  return excludeColors(globalColors)
+  return [
+    ...excludeColors(globalColors),
+    ...Object.keys(userColors)
```

Comment:
You should not export `$gray` or `$primary` here, otherwise it gets exported from `#ui-colors` (https://github.com/nuxt/ui/blob/dev/src/templates.ts#L4) and will be prompted to users in `color` prop: https://github.com/nuxt/ui/blob/dev/src/runtime/types/button.d.ts#L8

Reply:
Sure. Taking a look now.

Reply:
Not sure of the best way to handle this, or sufficiently confident I understand some of the subtleties of the code.

A simple option might be to filter the keys in the template creation function, but I am wondering if perhaps I misunderstood the role of `setGlobalColors()` and perhaps the `theme` object should not be augmented here.

If that was the case, then maybe in `src/tailwind.ts` we should detect and add `primary` and `gray` to `nuxt.options.appConfig.ui.themeColors` and refactor how `plugins/colors.ts` determines where to pull the colors from, referencing `ui.themeColors` instead.

I'll try this second option now, but if you want to offer any advice, I'm happy to hear it.

Reply:
I'm guessing, looking at the interface for app config, that perhaps app config is the wrong place to save the colors.

The challenge is caching `primary` and `gray` somewhere so they can be accessed in `plugins/colors.ts`.

Reply:
OK, think I have it using `runtimeConfig`.

It's a bit late now, but I'll submit a revised PR tomorrow.